tests/dataset_runner.py

```diff
@@ -228,6 +230,20 @@ def wait_for_primary_bundles(self):
             raise RuntimeError(f'Expected {self.expected_bundle_count} primary bundles, but only '
                                f'got {primary_bundles_count}')
 
+    def assert_valid_schema_versions_in_provenance(self):
+        primary_bundles = [self.data_store.bundle_manifest(bundle_uuid, "aws") for bundle_uuid in self.submission_envelope.bundles()]
+
+        metadata_files = []
+        for bundle in primary_bundles:
+            metadata_file_manifests = filter(lambda file: "metadata" in file["content-type"], bundle["bundle"]["files"])
+            metadata_files.extend([self.data_store.get_file(file["uuid"], "aws") for file in metadata_file_manifests])
+
+        for metadata_file in metadata_files:
+            schema_url = metadata_file["describedBy"]
+            schema = requests.get(schema_url).json()
+            validate(metadata_file, schema=schema)
```

Comment:
This validate function won't actually check to make sure that `schema` and the major and minor versions of the metadata file will match because in the schema, the pattern just says something like "^(http|https)://schema.(.*?)humancellatlas.org/type/biomaterial/(([0-9]{1,}.[0-9]{1,}.[0-9]{1,})|([a-zA-Z]*?))/donor_organism".

At least this is what I think. Have you verified to make sure this test fails if there is a mismatch?